bc_obps/reporting/service/report_review_changes_service.py

```diff
@@ -137,9 +138,84 @@ def process_change(
     @classmethod
     def get_report_version_diff_changes(cls, previous_data: dict, current_data: dict) -> List[Dict[str, Any]]:
         """
-        Compares previous and current report data using DeepDiff, filters out irrelevant changes,
-        and returns a list of processed changes with human-readable field names and values.
+        Compare previous and current report data using DeepDiff.
+        Handles facility_reports specially and returns a list of human-readable changes.
         """
+        changed: List[Dict[str, Any]] = []
+
+        # ---  if registration_purpose changed ---
+        prev_purpose = previous_data.get("report_operation", {}).get("registration_purpose")
+        curr_purpose = current_data.get("report_operation", {}).get("registration_purpose")
+        if prev_purpose != curr_purpose:
+            return [
+                {
+                    "field": "root['report_operation']['registration_purpose']",
+                    "old_value": prev_purpose,
+                    "new_value": curr_purpose,
+                    "change_type": "modified",
+                }
+            ]
+
+        # --- Special handling for facility_reports ---
+        if "facility_reports" in previous_data and "facility_reports" in current_data:
+            old_facilities = previous_data["facility_reports"]
+            new_facilities = current_data["facility_reports"]
+
+            removed = set(old_facilities.keys()) - set(new_facilities.keys())
+            added = set(new_facilities.keys()) - set(old_facilities.keys())
+
+            if len(removed) == 1 and len(added) == 1:
```

Comment:
is this the part that handles the name change of a facility?

I'm suspicious about the hard-coded value of 1. What if multiple facilities get renamed?